torchdata/datapipes/iter/util/unzipper.py

```diff
@@ -65,11 +65,15 @@ def __new__(
             )
 
         # The implementation basically uses Forker but only yields a specific element within the sequence
-        container = _UnZipperIterDataPipe(source_datapipe, sequence_length, buffer_size)  # type: ignore[arg-type]
-        return [_ChildDataPipe(container, i) for i in instance_ids]
+        container = _UnZipperIterDataPipe(source_datapipe, instance_ids, buffer_size)  # type: ignore[arg-type]
+        return [_ChildDataPipe(container, i) for i in range(len(instance_ids))]
 
 
 class _UnZipperIterDataPipe(_ForkerIterDataPipe):
+    def __init__(self, datapipe: IterDataPipe, instance_ids: List[int], buffer_size: int = 1000):
+        super().__init__(datapipe, len(instance_ids), buffer_size)
+        self.instance_ids = instance_ids
+
     def get_next_element_by_instance(self, instance_id: int):
         for return_val in super().get_next_element_by_instance(instance_id):
```

Comment:
super nit: Let's add a comment for the method here so that we know `return_val` is a tuple